dbt/adapters/bigquery/python_submissions.py

```diff
@@ -1,187 +1,163 @@
+
         if response.status.state == 6:
             raise ValueError(response.status.details)
+
         return response
 
 
-class ServerlessDataProcHelper(BaseDataProcHelper):
-    def _get_job_client(self) -> dataproc_v1.BatchControllerClient:
-        return dataproc_v1.BatchControllerClient(
-            client_options=self.client_options, credentials=self.GoogleCredentials
-        )
+class ServerlessDataProcHelper(_BaseDataProcHelper):
+    def __init__(self, parsed_model: Dict, credentials: BigQueryCredentials) -> None:
+        super().__init__(parsed_model, credentials)
+        self._batch_controller_client = batch_controller_client(credentials)
+        self._batch_id = parsed_model["config"].get("batch_id", str(uuid.uuid4()))
+        self._jar_file_uri = parsed_model["config"].get("jar_file_uri", _DEFAULT_JAR_FILE_URI)
+        self._dataproc_batch = credentials.dataproc_batch
 
-    def _get_batch_id(self) -> str:
-        model = self.parsed_model
-        default_batch_id = str(uuid.uuid4())
-        return model["config"].get("batch_id", default_batch_id)
-
-    def _submit_dataproc_job(self) -> Batch:
-        batch_id = self._get_batch_id()
-        logger.info(f"Submitting batch job with id: {batch_id}")
-        request = create_batch_request(
-            batch=self._configure_batch(),
-            batch_id=batch_id,
-            region=self.credential.dataproc_region,  # type: ignore
-            project=self.credential.execution_project,  # type: ignore
-        )
-        # make the request
-        self.job_client.create_batch(request=request)
-        return poll_batch_job(
-            parent=request.parent,
-            batch_id=batch_id,
-            job_client=self.job_client,
-            timeout=self.timeout,
+    def submit(self, compiled_code: str) -> Batch:
+        _logger.info(f"Submitting batch job with id: {self._batch_id}")
+
+        self._upload_to_gcs(compiled_code)
+
+        request = CreateBatchRequest(
+            parent=f"projects/{self._project}/locations/{self._region}",
+            batch=self._batch(),
+            batch_id=self._batch_id,
         )
-        # there might be useful results here that we can parse and return
-        # Dataproc job output is saved to the Cloud Storage bucket
-        # allocated to the job. Use regex to obtain the bucket and blob info.
-        # matches = re.match("gs://(.*?)/(.*)", response.driver_output_resource_uri)
-        # output = (
-        #     self.storage_client
-        #     .get_bucket(matches.group(1))
-        #     .blob(f"{matches.group(2)}.000000000")
-        #     .download_as_string()
-        # )
-
-    def _configure_batch(self):
+
+        # submit the batch
+        operation = self._batch_controller_client.create_batch(request)
+
+        # wait for the batch to complete
+        response: Batch = operation.result(polling=self._polling_retry)
+
+        return response
+
+    def _batch(self) -> Batch:
```

Comment:
nit: 
```suggestion
    def _get_batch_job_config(self) -> Batch:
```

Reply:
I think we get an actual Batch there, not just a config; though I suppose a Batch is actually a config itself. I'd also like to remove extra verbs from our method names. Get feels redundant here, to the point that this could even be a property if we wanted. But how about a compromise. I searched for different opinions and found [this](https://melevir.medium.com/python-functions-naming-the-algorithm-74320a18278d) one that I can get behind at an overall level (not just naming this method). I think it also addresses why I don't like `get`, which I frankly didn't realize why previously. So if we go with that, this should be `_create_batch(self) -> Batch`.

Reply:
I'm good with `_create_batch`